### PR TITLE
docs: Added many examples for on events

### PR DESCRIPTION
**Pull request by tristanperrow**
See [#477](https://github.com/kaplayjs/kaplay/pull/477) for more details.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1571,42 +1571,98 @@ export interface KAPLAYCtx<
...
     onDraw(action: () => void): KEventController;
     /**
-     * Register an event that runs when a game obj with certain tag is created.
-     *
-     * @param tag - The tag to listen for.
-     * @param action - The function to run when the event is triggered.
-     *
+     * Register an event that runs when an object with the provided tag is added.
+     * 
+     * @param tag - The tag for which the event runs.
```
Comment (lajbel) [inline src/types.ts]: The tag to listen for is what is most used

File: src/types.ts [hunk 2]
```diff
@@ -1571,42 +1571,98 @@ export interface KAPLAYCtx<
...
      */
     onAdd(tag: Tag, action: (obj: GameObj) => void): KEventController;
     /**
-     * Register an event that runs when a game obj is created.
-     *
-     * @param action - The function to run when the event is triggered.
-     *
+     * Register an event that runs when an object is added
+     * 
+     * @param action - The callback that runs when an object is added.
```
Comment (lajbel) [inline src/types.ts]: callback -> function

File: src/types.ts [hunk 3]
```diff
@@ -1571,42 +1571,98 @@ export interface KAPLAYCtx<
...
     onAdd(action: (obj: GameObj) => void): KEventController;
     /**
-     * Register an event that runs when a game obj with certain tag is destroyed.
-     *
-     * @param tag - The tag to listen for.
-     * @param action - The function to run when the event is triggered.
-     *
+     * Register an event that runs when an object with the provided tag is destroyed.
+     * 
+     * @param tag - The tag for which the event runs.
```
Comment (lajbel) [inline src/types.ts]: -> listen for

**Review by lajbel — CHANGES_REQUESTED** [review]

**Comment by tristanperrow** [comment]
Should be all fixed now

**Review by lajbel — APPROVED** [review]

**Comment by lajbel** [comment]
Great! Thank you so much

**Pull request merged**